### PR TITLE
[aws] Added more cluster configurations including GPUs, cleaned up CloudWatch, added more AWS tips and integrations

### PR DESCRIPTION
**Pull request by cfregly**
## Why are these changes needed?

Added more cluster configurations including GPUs, cleaned up CloudWatch, added more AWS tips and integrations

## Checks

- [X ] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [X] Release tests
   - [X] This PR is not tested :(

### CODE REVIEW

**Comment by cfregly** [comment]
@pdames Please review.

File: doc/source/cluster/cloud.rst [hunk 1]
```diff
@@ -45,10 +45,15 @@ Ray with cloud providers
         # Tear down the cluster.
         $ ray down ray/python/ray/autoscaler/aws/example-full.yaml
 
+    .. note::
+
+        See :ref:`aws-cluster` for additional AWS integrations and optimizations.
+
+        Refer to the `ray/python/ray/autoscaler/aws/example-gpu-docker.yaml <https://github.com/ray-project/ray/tree/master/python/ray/autoscaler/aws/example-gpu-docker.yaml>`__ cluster config file to setup a GPU cluster on AWS.
```
Comment (wuisawesome) [inline doc/source/cluster/cloud.rst:52]:
```suggestion
```

Seems a bit weird that we're providing a docker specific example here, redirecting them to the aws cluster page should be enough.

File: doc/source/cluster/aws-tips.rst [hunk 2]
```diff
@@ -5,39 +5,61 @@
...
+~~~~~~~~~~
+
+CPUs and GPUs
+=============
+
+Deploy Ray clusters on AWS using any combination of CPUs and GPUs.  
+
+
+.. code-block:: text
+
```
Comment (wuisawesome) [inline doc/source/cluster/aws-tips.rst:20]: Can you include the cluster name and provider fields so this becomes a fully copy/pastable example? (it should only add 3 lines to the config)

File: doc/source/cluster/aws-tips.rst [hunk 3]
```diff
@@ -5,39 +5,61 @@
...
 
-  You need to replace the ``{{FileSystemId}}`` to your own EFS ID before using the config. You may also need to set correct ``SecurityGroupIds`` for the instances in the config file.
+	Refer to the `ray/python/ray/autoscaler/aws/example-gpu-docker.yaml <https://github.com/ray-project/ray/tree/master/python/ray/autoscaler/aws/example-gpu-docker.yaml>`__ cluster config file to setup a GPU cluster on AWS.
 
-.. code-block:: yaml
+Spot Instances
+==============
+
+.. code-block:: text
+
```
Comment (wuisawesome) [inline doc/source/cluster/aws-tips.rst:43]: Same here, fully copy/pastable pls

File: doc/source/cluster/aws-tips.rst [hunk 4]
```diff
@@ -73,10 +95,199 @@ You should see something like
 Please refer to `this discussion <https://github.com/ray-project/ray/issues/9327>`__ for more details.
 
 
+.. _aws-cluster-efs:
+
+Amazon EFS
+~~~~~~~~~~
+
+To use Amazon EFS, install some utilities and mount the EFS in ``setup_commands``. Note that these instructions only work if you are using the AWS Autoscaler.
```
Comment (wuisawesome) [inline doc/source/cluster/aws-tips.rst:103]: Is there some EFS documentation we can link to here about what utilities we're installing and why? maybe this page? https://docs.aws.amazon.com/efs/latest/ug/installing-amazon-efs-utils.html#installing-other-distro

File: doc/source/cluster/aws-tips.rst [hunk 5]
```diff
@@ -73,10 +95,199 @@ You should see something like
...
+            sudo apt-get -y install ./build/amazon-efs-utils*deb;
+            cd $HOME;
+            mkdir efs;
+            sudo mount -t efs {{FileSystemId}}:/ efs;
+            sudo chmod 777 efs;
+
+
+.. _aws-cluster-emr:
+
+Amazon EMR
```
Comment (wuisawesome) [inline doc/source/cluster/aws-tips.rst:130]: Can you remove this and the following sagemaker section? These are different deployment models. 

It's not enough for these examples to just work at a given point in time. Adding them to the docs creates an implication of supporting those deployment models, which is a rather large decision.

AWS can provide its own support for running Ray on EMR, but Ray's would only be providing first-class support via the manual cluster setup steps (e.g. the public `ray start`/`ray stop` apis). 

If you feel strongly about trying to get first class support for EMR, I think the right next steps would be to open up a [REP (ray enhancement proposal) ](https://github.com/ray-project/enhancements)

**Review by wuisawesome — CHANGES_REQUESTED** [review]

**Review by wuisawesome — CHANGES_REQUESTED** [review]
Thanks for the contribution @cfregly!

Left a couple of comments that fall under 3 categories.
1. The EMR and SageMaker sections are different deployment models with large implications and we need to go through a more thorough process for those.
2. In general, users should ideally be able to copy/paste the examples and they should be fully working.
3. A few other minor formatting comments.

**Pull request closed without merging**